Quake/r_world.c

```diff
@@ -585,8 +604,90 @@ void R_DrawTextureChains_Water (qmodel_t *model, entity_t *ent, texchain_t chain
 			R_EndTransparentDrawing (entalpha);
 		}
 	}
+	else if (cl.worldmodel->haslitwater && r_litwater.value)
```

Comment:
I think this also needs `&& r_world_program != 0` - that's how `R_DrawTextureChains` checks before using the GLSL codepath.

Reply:
Nice, added and confirmed that without this I would in fact get a segfault if I run with `-noglsl`.